examples/cpp/CGroupTest.cc

```diff
@@ -0,0 +1,79 @@
+/*
+ * CPUDistribution Show load distribution across CPU cores during a period of
+ *                 time. For Linux, uses BCC, eBPF. Embedded C.
+ *
+ * Basic example of BCC and kprobes.
+ *
+ * USAGE: CPUDistribution [duration]
+ *
+ * Copyright (c) Facebook, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ */
+
+#include <unistd.h>
+#include <fstream>
+#include <cstdlib>
+#include <iomanip>
+#include <iostream>
+#include <string>
+
+#include "BPF.h"
+
+const std::string BPF_PROGRAM = R"(
+#include <linux/sched.h>
+#include <linux/path.h>
+#include <linux/dcache.h>
+
+BPF_CGROUP_ARRAY(cgroup, 1);
+
+int on_vfs_open(struct pt_regs *ctx, struct path *path) {
+  if (cgroup.check_current_task(0))
+    bpf_trace_printk("file '%s' was opened!\n", path->dentry->d_name.name);
+  return 0;
+}
+)";
+
+int main(int argc, char** argv) {
+  ebpf::BPF bpf;
+  auto init_res = bpf.init(BPF_PROGRAM);
+  if (init_res.code() != 0) {
+    std::cerr << init_res.msg() << std::endl;
+    return 1;
+  }
+
+  if (argc == 2) {
+    auto cgroup_array = bpf.get_cgroup_array("cgroup");
+    auto update_res = cgroup_array.update_value(0, argv[1]);
+    if (update_res.code() != 0) {
+      std::cerr << update_res.msg() << std::endl;
+      return 1;
+    }
+  }
+
+  auto attach_res =
+      bpf.attach_kprobe("vfs_open", "on_vfs_open");
+  if (attach_res.code() != 0) {
+    std::cerr << attach_res.msg() << std::endl;
+    return 1;
+  }
+
+  std::ifstream pipe("/sys/kernel/debug/tracing/trace_pipe");
+  std::string line;
+
+  while (true) {
+    if (std::getline(pipe, line)) {
+      std::cout << line << std::endl;
+    } else {
+      std::cout << "Waiting for a sys_clone event" << std::endl;
+      sleep(1);
+    }
+  }
```

Comment:
Looks like we have no way to get out of the while loop. For an example, it is okay. If this is the intention, maybe you want to print out some thing like
```
std::cout << "Started tracing, hit Ctrl-C to terminate." << std::endl;
```
and you do not need anything after this loop.

Reply:
Yes, I know kernel will clean everything up after the process exits, but let's keep the code after this loop just for completeness.